lang/java/reef-utils/src/main/java/org/apache/reef/util/Optional.java

```diff
@@ -35,19 +35,15 @@
   private static final long serialVersionUID = 42L;
 
   private final T value;
-  private final String valueStr;
-  private final int valueHash;
+  private static final String EMPTY_VALUE_STR = "OptionalvNothing";
```

Comment:
Referring to the [Java8 doc](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#toString--), it is recommended that

> If a value is present the result must include its string representation in the result. Empty and present Optionals must be unambiguously differentiable

and this was how it was implemented originally. Although your suggestion has the same format, maybe the curly braces makes it too similar to the result with present `Options`?
Digging in deeper, maybe we could borrow openJDK 8's implementation, and set it as `"Optional.empty"`.